source/Lexer.cpp

```diff
@@ -129,32 +131,29 @@ Token Lexer::concatenateTokens(BumpAllocator& alloc, Token left, Token right) {
     mem[newLength - 1] = '\0';
     StringRef combined{ (char*)mem, newLength };
 
-    // common case: identifier + identifier
```

Comment:
Why did you get rid of this case? It's an optimization that will probably be true in most cases.